integration-auth/src/main/java/org/symphonyoss/integration/authentication/AuthenticationContext.java

```diff
@@ -50,15 +66,47 @@
 
   public AuthenticationContext(String userId, KeyStore keyStore, String keyStorePass) {
     this.userId = userId;
+    this.client = buildClient(keyStore, keyStorePass);;
+  }
 
+  private Client buildClient(KeyStore keyStore, String keyStorePass) {
```

Comment:
We took a few measures on the client instantiation to avoid memory leaks. Put read and connect timeouts and also configured a pool of connections. For the tests I did, the upgrade of Jersey version and closing responses where not effective to avoid the memory leak. The timeouts on the API client were effective, and the connection pool was really something we did as we were already touching this point (it is just for performance improvement really).